### PR TITLE
DexScreener v25 backport

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

DexScreener prototype by @cryptomatictrader . Opening it up as draft PR so that we can start discussing.

### CODE REVIEW

File: ingest/sqs/service/sqs_streaming_service.go [hunk 1]
```diff
@@ -69,6 +70,24 @@ func (s *sqsStreamingService) ListenDeliverTx(ctx context.Context, req types.Req
...
+	block := indexerdomain.Block{
+		ChainId:     chainId,
+		Height:      height,
+		BlockTime:   timeEndBlock,
+		GasConsumed: gasConsumed,
+	}
+	err := s.sqsIngester.PublishBlock(sdk.UnwrapSDKContext(ctx), block)
+	if err != nil {
+		return err
+	}
```
Comment (p0mvn) [inline ingest/sqs/service/sqs_streaming_service.go:89]: We will need to remove this before merge

File: ingest/sqs/ingester.go [hunk 2]
```diff
@@ -13,22 +14,32 @@ var _ domain.Ingester = &sqsIngester{}
...
 	}
 }
 
+func (i *sqsIngester) PublishBlock(ctx sdk.Context, block indexerdomain.Block) error {
+	err := i.indexerPubSubClient.PublishBlock(ctx, block)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (p0mvn) [inline ingest/sqs/ingester.go:41]: We will need to move this to the indexer ingester as well pre-merge

File: ingest/indexer/service/writelistener/bank_write_listener.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		denom := key[len(banktypes.SupplyKey):]
+		if err := s.publishSupply(denom, value); err != nil {
+			return err
+		}
+	}
+
+	// If the cold start manager has ingested initial data and the key is not empty and the key is a supply key.
+	if len(key) > 0 && bytes.Equal(banktypes.SupplyOffsetKey, key[:1]) {
+		denom := key[len(banktypes.SupplyOffsetKey):]
+		if err := s.publishSupply(denom, value); err != nil {
```
Comment (p0mvn) [inline ingest/indexer/service/writelistener/bank_write_listener.go:55]: Note: we will need to create a separate function for supply offsets in a followup PR

(I did not have time to finish it in the original PR)

Comment (p0mvn) [reply ingest/indexer/service/writelistener/bank_write_listener.go:55]: Followed-up here: https://github.com/osmosis-labs/osmosis/pull/8404

**Review by cryptomatictrader — APPROVED** [review]

**Pull request merged**